src/main/java/graphql/normalized/ExecutableNormalizedOperationFactory.java

```diff
@@ -188,14 +248,39 @@ public static ExecutableNormalizedOperation createExecutableNormalizedOperation(
                                                                                     OperationDefinition operationDefinition,
                                                                                     Map<String, FragmentDefinition> fragments,
                                                                                     CoercedVariables coercedVariableValues) {
-        return new ExecutableNormalizedOperationFactory().createNormalizedQueryImpl(graphQLSchema,
+        return createExecutableNormalizedOperation(graphQLSchema,
                 operationDefinition,
                 fragments,
                 coercedVariableValues,
-                null,
                 Options.defaultOptions());
     }
 
+    /**
+     * This will create a runtime representation of the graphql operation that would be executed
+     * in a runtime sense.
+     *
+     * @param graphQLSchema         the schema to be used
+     * @param operationDefinition   the operation to be executed
+     * @param fragments             a set of fragments associated with the operation
+     * @param coercedVariableValues the coerced variables to use
+     *
+     * @return a runtime representation of the graphql operation.
+     */
+    public static ExecutableNormalizedOperation createExecutableNormalizedOperation(GraphQLSchema graphQLSchema,
+                                                                                    OperationDefinition operationDefinition,
+                                                                                    Map<String, FragmentDefinition> fragments,
+                                                                                    CoercedVariables coercedVariableValues,
+                                                                                    Options options) {
+        return new ExecutableNormalizedOperationFactoryImpl(
```

Comment:
Structural change - a `ExecutableNormalizedOperationFactoryImpl` was added after v21.x, in preparation for defer